test/acceptance/test/utils.go

```diff
@@ -270,15 +279,54 @@ func createGitRepoBranch(repoAbsolutePath string, branchName string) string {
 	return string(session.Wait().Out.Contents())
 }
 
-func getRepoVisibility(org string, repo string) string {
-	command := exec.Command("sh", "-c", fmt.Sprintf("hub api --flat repos/%s/%s|grep -i private|cut -f2", org, repo))
-	session, err := gexec.Start(command, GinkgoWriter, GinkgoWriter)
+func getGitHubRepoVisibility(org string, repo string) string {
+	gitProvider, err := github.NewClient(
+		gitprovider.WithOAuth2Token(os.Getenv("GITHUB_TOKEN")),
+		gitprovider.WithDestructiveAPICalls(true),
+	)
 	Expect(err).ShouldNot(HaveOccurred())
-	Eventually(session).Should(gexec.Exit())
-	visibilityStr := strings.TrimSpace(string(session.Wait().Out.Contents()))
-	log.Infof("Repo visibility private=%s", visibilityStr)
 
-	return visibilityStr
+	orgRef := gitprovider.OrgRepositoryRef{
+		RepositoryName: repo,
+		OrganizationRef: gitprovider.OrganizationRef{
+			Domain:       github.DefaultDomain,
+			Organization: GITHUB_ORG,
+		},
+	}
+
+	orgInfo, err := gitProvider.OrgRepositories().Get(context.Background(), orgRef)
+	Expect(err).ShouldNot(HaveOccurred())
+
+	visibility := string(*orgInfo.Get().Visibility)
+	log.Infof("Repo visibility private=%s", visibility)
+
+	return visibility
+}
+
+func getGitLabRepoVisibility(org string, repo string) string {
+	gitProvider, err := gitlab.NewClient(
+		os.Getenv("GITLAB_TOKEN"),
+		"oauth2",
+		gitprovider.WithOAuth2Token(os.Getenv("GITLAB_TOKEN")),
+		gitprovider.WithDestructiveAPICalls(true),
+	)
+	Expect(err).ShouldNot(HaveOccurred())
+
+	orgRef := gitprovider.OrgRepositoryRef{
+		RepositoryName: repo,
+		OrganizationRef: gitprovider.OrganizationRef{
+			Domain:       gitlab.DefaultDomain,
+			Organization: GITLAB_ORG,
+		},
+	}
+
+	orgInfo, err := gitProvider.OrgRepositories().Get(context.Background(), orgRef)
+	Expect(err).ShouldNot(HaveOccurred())
+
+	visibility := string(*orgInfo.Get().Visibility)
+	log.Infof("Repo visibility private=%s", visibility)
```

Comment:
It's weird to me that we log things here. I'm assuming that we log things elsewhere in acceptance tests?

Reply:
We are checking for this in the acceptance test.

Reply:
Sorry not understanding. We check that something gets logged?

Reply:
ohh sorry I led you wrong. We use check the result of this function not the actual log. You are right though it does seem weird that we log it.